src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs

```diff
@@ -55,5 +90,15 @@ private static IServiceCollection TrySetProductInfo(this IServiceCollection serv
             var productInfo = ProductInfo.GetProductInfo(assembly);
             return services.Configure<ServiceManagerContext>(o => o.ProductInfo = o.ProductInfo ?? productInfo);
         }
+
+        private static IServiceCollection AddRestClientFactory(this IServiceCollection services) => services
+            .AddHttpClient()
+            .AddSingleton(sp =>
+            {
+                var options = sp.GetRequiredService<IOptions<ServiceManagerContext>>().Value;
+                var productInfo = options.ProductInfo;
+                var httpClientFactory = sp.GetRequiredService<IHttpClientFactory>();
+                return new RestClientFactory(productInfo, httpClientFactory);
```

Comment:
change ctor to `RestClientFactory(IOptions<ServiceManagerContext> o, IHttpClientFactory)`?
and then `AddSingleton<RestClientFactory>()`

Reply:
As `RestClientFactory` is in `common` project, and `ServiceManagerContext` is `management` project specified, I think we'd better not introduce `ServiceManagerContext` into `RestClientFactory` .